worker/provisioner/provisioner_task_test.go

```diff
@@ -282,3 +362,51 @@ func (m *testMachine) MarkForRemoval() error {
 	m.markForRemoval = true
 	return nil
 }
+
+func (m *testMachine) Tag() names.Tag {
```

Comment:
Yes, we need these methods as the provisioner task calls this methods during the tests run. At this point, we do not need to assert anything here as the added tests have only limited coverage. This is just the first step for transferring the rest of the tests and increasing the coverage. Note that this approach let's us improve test coverage as it was difficult to exercise some code paths previously, with JujuConnSuite approach.
Extra test will be grown and transformed from existing either when the need arises, i.e. existing test in JujuConnSuite fails, or as an in-our-spare-time exercise :) say during sprint, in the airport, etc..

Reply:
To clarify - the removed TestProvisionerRetriesTransientErrors test is not fully replaced yet?
Can we leave a TODO to explain what coverage is missing so that it's clear what needs to be added back in? Ideally this would happen sooner rather than later.

Reply:
No coverage is missing for provisioner task. Whatever we currently have, we have. The 2 unit tests that have been moved, provide the same coverage as befrore.